src/twisted/web/http.py

```diff
@@ -1185,6 +1185,7 @@ def write(self, data):
                 and (self.responseHeaders.getRawHeaders(b"content-length") is None)
                 and self.method != b"HEAD"
                 and self.code not in NO_BODY_CODES
+                and (self.method != b"CONNECT" and self.code // 100 == 2)
```

Comment:
"successful connect" is expressed twice here; it would be nice to unify that into a helper function or private method.